design/ceph/object/ceph-bucket-notification-crd.md

```diff
@@ -0,0 +1,91 @@
+kind: CephBucketTopic
+metadata:
+  name: # name of the topic
+  namespace: # namespace where topic belongs
+Spec:
+  endpoint: #(mandatory) URI of an endpoint to send push notification to
+  opaqueData: #(optional) opaque data is set in the topic configuration
+  persistent: false #(optional) indication whether notifications to this endpoint are persistent or not (`false` by default)
+  # Endpoint specific parameters
+  http:
+    verifySSL: true #indicate whether the server certificate is validated by the client or not (`true` by default)
+  amqp:
+    ackLevel: broker # none/routable/broker, optional (default - broker)
+    amqpExchange: direct # exchanges must exist and be able to route messages based on topics
+  kafka:
+    useSSL: true # secure connection will be used for connecting with the broker (`false` by default)
+    caLocation: <filepath in rgw pod> # this specified CA will be used, instead of the default one, to authenticate the broker
+    ackLevel: broker # none/broker, optional (default - broker)
+```
+P.S : Endpoint can be of different format depends on the server
+- http -> `http[s]://<fqdn>[:<port]`
+- amqp -> `amqp://[<user>:<password>@]<fqdn>[:<port>][/<vhost>]`
+- kafka -> `kafka://[<user>:<password>@]<fqdn>[:<port]`
+
+The CR for bucket notification takes this form:
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephBucketNotification
+metadata:
+  name: # name of the notification
+  namespace: # namespace where notification belongs
+Spec:
+  topic: #(mandatory) reference to the topic, topic_arn
+  filter: #(optional) Prefix/Suffix/Regex/Metadata/Tags, optional (default - {})
+    metadata:
+    - name: x-amz-meta-color
+      value: blue
+    stringMatch:
+    - name: prefix
+      value: hello
+    - name: suffix
+      value: .png
+    - name: regex
+      value: [a-z]*
+
+  events: # applicable values [here](https://docs.ceph.com/en/latest/radosgw/s3-notification-compatibility/#event-types), (default all)
+```
+
+After creating resources for notifications, it can be referred in `Additional Config` field of OBC. `Additional Config` is string map of key value pairs and notification can be referred as `bucketNotification` key. It will be list of notification separated by comma.
+
+```yaml
+apiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucketClaim
+metadata:
+  name: ceph-bucket
+spec:
+  bucketName: mybucket
+  storageClassName: rook-ceph-delete-bucket
+  additionalConfig:
+    bucketNotification: "bucketnotification1, bucketnotification2"
+```
+
+```
+P.S Currently OBC's CRDs cannot be modified/updated after creation please [see](https://github.com/kube-object-storage/lib-bucket-provisioner#195), hence bucket notifications could be added only upon the creation of a bucket, and cannot be removed
+```
+
+## Out of Scope
+Upcoming [COSI design](https://github.com/kubernetes/enhancements/pull/1383/) does not have `Additional Config` field in its CRD. So the `bucket notification` may need to handle differently.
```

Comment:
The enhancement has merged, so you can link to the [design doc](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1979-object-storage-support) directly now instead of the PR.

Reply:
Going to update this link?